srtcore/threadname.h

```diff
@@ -62,9 +97,14 @@ class ThreadName
             // null byte. (If the length of the string, including the terminating
             // null byte, exceeds 16 bytes, the string is silently truncated.)
             return prctl(PR_SET_NAME, (unsigned long)name, 0, 0) != -1;
-#elif defined(__APPLE__)
-            // since macos(10.6), ios(3.2)
-            return pthread_setname_np(name) == 0;
+#elif defined(HAVE_PTHREAD_SETNAME_NP)
+            #if defined(__APPLE__)
+            return name != NULL
+               && pthread_setname_np(name) == 0;
```

Comment:
I would rather put an assertion at the beginning of the function, instead of checking in every ifdef case.
The function is used internally by SRT, and a null pointer is not expected and must not be silenced by returning `false`.
```c++
SRT_ASSERT(name != NULL);
```

Reply:
The problem with using `assert` is that in non-debug builds, the macro is defined to generate no code. I can make the change if you wish, but maybe return false on `name == NULL` and have the assert in place as well. One thing I dont like about that is that the behavior will be different in debug and non-debug builds. Which never seems right to me.

Reply:
Do we have any premise to believe that on particular platform and compiled version, this NULL will happen either always or never?

Reply:
Well it is a parameter passed to the method. The intent here is that if the method is called with a `name == NULL` we get a consistent result no matter the underlying implementation of `pthread_setname_np()` considering the standard does not specify what will happen other than that the standard expects the `name` parameter to be a valid C string. Since `NULL` is not, this method does not try to call `pthread_setname_np()` with a `NULL` parameter, but returns `false` instead.

Reply:
`ThreadName::set(const char* name)` is used by SRT internally, and `NULL` must not be passed as an argument. And it is not as of now. Therefore I believe there is no need to check the value in runtime. If `NULL` is provided, it is the development error that must be fixed.